### PR TITLE
extract a flutter utility method

### PR DESCRIPTION
**Pull request by pq**
/cc @bwilkerson

### CODE REVIEW

File: lib/src/util/flutter_utils.dart [hunk 1]
```diff
@@ -61,13 +66,38 @@ class _Flutter {
         _uriContainer = Uri.parse('$uriPrefix/src/widgets/container.dart'),
         _uriFramework = Uri.parse('$uriPrefix/src/widgets/framework.dart');
 
-  bool isExactWidgetTypeContainer(DartType? type) =>
-      type is InterfaceType &&
-      _isExactWidget(type.element, _nameContainer, _uriContainer);
+  bool hasWidgetAsAscendant(ClassElement? element,
+      [Set<ClassElement>? alreadySeen]) {
+    alreadySeen ??= {};
+    if (element == null || alreadySeen.contains(element)) {
```
Comment (bwilkerson) [inline lib/src/util/flutter_utils.dart:72]: `Set.add` returns `true` if the set was modified, so consider using `!add` in place of `contains`. It's more performant that way.

File: lib/src/util/flutter_utils.dart [hunk 2]
```diff
@@ -61,13 +66,38 @@ class _Flutter {
...
+  }
+
+  bool isBuildContext(DartType? type, {bool skipNullable = false}) {
+    if (type is! InterfaceType) {
+      return false;
+    }
+    if (skipNullable && type.nullabilitySuffix == NullabilitySuffix.question) {
+      return false;
+    }
+    var element = type.element;
```
Comment (bwilkerson) [inline lib/src/util/flutter_utils.dart:89]: Consider using `_isExactWidget` here.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/util/flutter_utils.dart (hunk 2 above)
Comment (pq) [reply lib/src/util/flutter_utils.dart:89]: Oh, thanks!  I forgot I'd inlined that.

File: lib/src/util/flutter_utils.dart (hunk 1 above)
Comment (pq) [reply lib/src/util/flutter_utils.dart:72]: Nice one.  Thanks!

**Pull request merged**